project/src/main/puzzle/piece/piece-queue.gd

```diff
@@ -9,11 +9,15 @@ This queue stores the upcoming pieces so they can be displayed, and randomizes t
 # minimum number of next _pieces in the queue, before we add more
 const MIN_SIZE := 50
 
+const UNLIMITED_PIECES := 999999
+
 var _pieces := []
 
 # default pieces to pull from if none are provided by the level
 var _default_piece_types := PieceTypes.all_types
 
+var _remaining_piece_count := UNLIMITED_PIECES
```

Comment:
This value is never decremented, so it will not work for larger values.

Reply:
Value is now decremented properly. I tested it with a piece queue size of 20, and verified it was broken before, and works OK now.